src/Language/PureScript/Bundle.hs

```diff
@@ -145,13 +145,20 @@ printErrorMessage (MissingMainModule mName) =
 
 -- | Calculate the ModuleIdentifier which a require(...) statement imports.
 checkImportPath :: String -> ModuleIdentifier -> S.Set String -> Either String ModuleIdentifier
-checkImportPath "./foreign" m _ =
+checkImportPath "./foreign.js" m _ =
   Right (ModuleIdentifier (moduleName m) Foreign)
 checkImportPath name _ names
-  | Just name' <- stripPrefix "../" name
+  | Just name' <- stripSuffix "/index.js" =<< stripPrefix "../" name
   , name' `S.member` names = Right (ModuleIdentifier name' Regular)
 checkImportPath name _ _ = Left name
 
+stripSuffix :: Eq a => [a] -> [a] -> Maybe [a]
```

Comment:
Could this just be: https://hackage.haskell.org/package/filepath-1.4.2/docs/System-FilePath-Posix.html#v:takeDirectory?

Reply:
I guess it doesn't do the `Maybe` thing...

Reply:
Perhaps, yeah - but this way at least ensures it's exactly the expected pattern, whereas that might be too broad, unless we check the filename part separately too.